YOZM/YOZM/Model/CloudKit/ChapterCloudKit.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  ChapterCloudKit.swift
+//  YOZM
+//
+//  Created by 최희진 on 9/2/25.
+//
+
+import CloudKit
+
+struct ChapterCloudKit {
+    let record: CKRecord
+
+    func id() throws -> Int64 {
+        guard let value = record[CloudKitField.id.rawValue] as? Int64 else {
+            throw CloudKitError.missingField(field: CloudKitField.id.rawValue)
+        }
+        return value
+    }
+
+    func title() throws -> String {
+        guard let value = record[CloudKitField.title.rawValue] as? String else {
+            throw CloudKitError.missingField(field: CloudKitField.title.rawValue)
+        }
+        return value
+    }
+
+    init(id: Int64, title: String) {
+        self.record = CKRecord(recordType: CloudKitType.chapterRecordType)
+        self.record[CloudKitField.id.rawValue] = id as CKRecordValue
+        self.record[CloudKitField.title.rawValue] = title as CKRecordValue
+    }
+
+    init(record: CKRecord) {
+        self.record = record
+    }
+}
```

Comment:
`record` 프로퍼티 대신 `id`, `title` 프로퍼티로 저장하고 `toModel() -> Chapter`처럼 함수로 바로 변환할 수 있게 하는 것은 어떨까요?

Reply:
```suggestion
struct ChapterCloudKit {
    let id: Int64
    let title: String

    init(id: Int64, title: String) {
        self.id = id
        self.title = title
    }
}

extension ChapterCloudKit{
    func toCKRecord() -> CKRecord {
        let record = CKRecord(recordType: CloudKitType.chapterRecordType)
        record[CloudKitField.id.rawValue] = id as CKRecordValue
        record[CloudKitField.title.rawValue] = title as CKRecordValue
        return record
    }
}
```

id와 title 프로퍼티로 저장하고 toCkRecord() 함수를 통해 CKRecord로 받을 수 있도록 수정하면 좋겠네요!
이전에는 ChapterCloudKit이 record 프로퍼티를 가지면서 CKRecord 보관뿐만 아니라 CKRecord 생성 책임과 같은 역할을 가져서 SRP를 지키지 못 한 것 같아요

그리고 네이밍은 `ChapterRecord`로 수정하는게 더 명확할 것 같은데 어떤가요?

Reply:
네 좋아요!

Reply:
수정했습니다!